src/PhpBrew/VariantBuilder.php

```diff
@@ -561,6 +562,39 @@ public function __construct()
             }
         };
 
+        /*
+         * --with-sodium option
+         *
+         * --with-sodium=shared
+         * --with-sodium=[dir]
+         *
+         * On ubuntu you need to install libsodium-dev
+         */
+        $this->variants['sodium'] = function (Build $build, $val = null) {
+            if ($val) {
+                return "--with-sodium=$val";
+            }
+
+            if ($prefix = Utils::findIncludePrefix('sodium.h')) {
+                return "--with-sodium=$prefix";
+            }
+
+            if ($prefix = Utils::getPkgConfigPrefix('sodium')) {
+                return "--with-sodium=$prefix";
+            }
+
+            $possiblePrefixes = array('/usr/local/opt/sodium');
```

Comment:
If this is expected to be created by HomeBrew, we should use the `BrewPrefixFinder` instead of hard-coding the path.